src/app/Dashboard/Topic.js

```diff
@@ -1,18 +1,53 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import './Topic.css';
 
-export default function Topic({ title, topicId }) {
-  return (
-    <button
-      className="topic-btn"
-      onClick={() => console.log('click through to topic/:id', topicId)}
-    >
-      {title}
-    </button>
-  );
-}
+export default class Topic extends React.Component {
+  static propTypes = {
+    topicId: PropTypes.number.isRequired,
+    title: PropTypes.string.isRequired
+  };
+
+  state = {
+    isHidden: true
+  };
+
+  editTopic(topicId) {
+    console.log('edit:', topicId);
+  }
+
+  deleteTopic(topicId) {
+    console.log('delete:', topicId);
+  }
 
-Topic.propTypes = {
-  topicId: PropTypes.number,
-  title: PropTypes.string
-};
+  toggleHidden() {
+    this.setState({
+      isHidden: !this.state.isHidden
+    });
+  }
+
+  render() {
+    const { title, topicId } = this.props;
+
+    return (
+      <div
+        className="topic-wrap"
+        onMouseEnter={() => this.toggleHidden()}
```

Comment:
Let's agree on the convention of referencing event callbacks directly using `this.{{callback}}`. This means that all class methods should be defined as arrow functions. This reduces weird bugs anyway, I promise! :D